mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -239,6 +242,11 @@ private boolean textContains(String value, MailboxMessage message) throws IOExce
         return isInMessage(value, new SequenceInputStream(textHeaders(message), bodyContent), true);
     }
 
+    private boolean atachmentContains(String value, MailboxMessage message) throws IOException, MimeException {
+        final InputStream input = message.getFullContent();
+        return isInMessage(value, input, true);
```

Comment:
It is not the strict definition of "in attachments". Maybe it is enough for the tests.

My idea would have been to look in *MailboxMessage attachments** which is then possible only if MessageManager supports ATTACHMENT capability.